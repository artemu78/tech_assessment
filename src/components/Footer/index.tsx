import React from "react";
export const links: [string, string][] = [
  ["questions source", "https://github.com/Ebazhanov/linkedin-skill-assessments-quizzes"],
  ["this app source repository", "https://github.com/artemu78/tech_assessment"],
  ["this app typescript coverage", "https://artemu78.github.io/tech_assessment/coverage-ts/"],
  ["this app test coverage", "https://artemu78.github.io/tech_assessment/coverage/"],
  ["hire/contact me", "https://t.me/artemu78"],
];

const Footer = (): JSX.Element => {
  return (
    <footer style={{ backgroundColor: "lightgrey" }}>
      {links.map((link, index) => {
        return (
          <span key={index + link[1]}>
            <a href={link[1]}>{link[0]}</a>
            {links.length - 1 !== index ? ", " : ""}
          </span>
        );
      })}
    </footer>
  );
};

export default Footer;
