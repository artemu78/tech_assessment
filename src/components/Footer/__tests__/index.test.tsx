import React from "react";
import { render, screen } from "@testing-library/react";
// import renderer from "react-test-renderer";
import Footer, { links } from "..";

test("render Footer", () => {
  const { container } = render(<Footer />);

  links.forEach((link) => {
    expect(screen.getByText(link[0])).toHaveAttribute("href", link[1]);
  });
  expect(container).toMatchSnapshot();
});
