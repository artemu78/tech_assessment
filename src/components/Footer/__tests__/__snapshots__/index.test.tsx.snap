// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Footer 1`] = `
<div>
  <footer
    style="background-color: lightgrey;"
  >
    <span>
      <a
        href="https://github.com/Ebazhanov/linkedin-skill-assessments-quizzes"
      >
        questions source
      </a>
      , 
    </span>
    <span>
      <a
        href="https://github.com/artemu78/tech_assessment"
      >
        this app source repository
      </a>
      , 
    </span>
    <span>
      <a
        href="https://artemu78.github.io/tech_assessment/coverage-ts/"
      >
        this app typescript coverage
      </a>
      , 
    </span>
    <span>
      <a
        href="https://artemu78.github.io/tech_assessment/coverage/"
      >
        this app test coverage
      </a>
      , 
    </span>
    <span>
      <a
        href="https://t.me/artemu78"
      >
        hire/contact me
      </a>
    </span>
  </footer>
</div>
`;
